### PR TITLE
Feature/graphql

### PR DESCRIPTION
**Pull request by ChrisMeyer7088**
# Description

Completely remodeled logging
Added queries using graphql

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

All unit tests and integration tests pass, still need to deploy test it

# Notes

### CODE REVIEW

File: Backend/WebApi-PointMap/Controllers/PointController.cs [hunk 1]
```diff
@@ -12,161 +12,163 @@
...
+			try
+			{ 
+				var pointId = ControllerHelpers.ParseAndCheckId(guid);     
+				var token = ControllerHelpers.GetToken(Request);
+				session = ControllerHelpers.ValidateAndUpdateSession(_db, token);
                 Guid id = new Guid(guid);
 
-                var point = _pm.GetPoint(_db, id);
-                _db.SaveChanges();
+				var point = _pm.GetPoint(_db, id);
```
Comment (alfredovargasjr) [inline Backend/WebApi-PointMap/Controllers/PointController.cs]: this formatting went a bit crazy haha

**Comment by alfredovargasjr** [comment]
There are conflicts and failing tests

File: Backend/DataAccessLayer/Database/dbContext.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace DataAccessLayer.Database
     public class DatabaseContext : DbContext
     {
         // TODO: turn into enviornmental variables for dev and deploy
+        readonly string productionDB = "Data Source=pointmapdbinstance.cqugps36mcx5.us-east-2.rds.amazonaws.com;Initial Catalog=pointmapdbinstance;User ID=Administrator;Password=NightWatch_123";
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Database/dbContext.cs]: was this left accidentally?

File: Backend/ManagerLayer/ManagerLayer.csproj [hunk 3]
```diff
@@ -52,8 +52,11 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="KFC_SSO_Utility\KFC_SSO_Manager.cs" />
+<<<<<<< HEAD
```
Comment (alfredovargasjr) [inline Backend/ManagerLayer/ManagerLayer.csproj]: unresolved conflict

File: Backend/WebApi-PointMap/ErrorHandling/GeneralErrorHandler.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+using System.Web;
+using static ServiceLayer.Services.ExceptionService;
+using System.ComponentModel.DataAnnotations;
+
+namespace WebApi_PointMap.ErrorHandling
+{
+    public class GeneralErrorHandler
+    {
+        public static HttpResponseMessage HandleException(Exception e)
+        {
```
Comment (alfredovargasjr) [inline Backend/WebApi-PointMap/ErrorHandling/GeneralErrorHandler.cs]: we dont have this anymore

File: Backend/WebApi-PointMap/WebApi-PointMap.csproj [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ﻿<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.1\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props" Condition="Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.1\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props')" />
+  <Import Project="..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.0\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props" Condition="Exists('..\packages\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.2.0.0\build\net46\Microsoft.CodeDom.Providers.DotNetCompilerPlatform.props')" />
```
Comment (alfredovargasjr) [inline Backend/WebApi-PointMap/WebApi-PointMap.csproj]: theres a update for webapi

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Backend/DataAccessLayer/Database/dbContext.cs (hunk 2 above)
Comment (ChrisMeyer7088) [reply Backend/DataAccessLayer/Database/dbContext.cs]: Just having it while i'm still testing this branch to make it easy to deploy, will remove once I'm ready to merge

File: Backend/WebApi-PointMap/ErrorHandling/GeneralErrorHandler.cs (hunk 4 above)
Comment (ChrisMeyer7088) [reply Backend/WebApi-PointMap/ErrorHandling/GeneralErrorHandler.cs]: Yup the physical files wasn't removed in my pull, but it was removed from the project

File: Backend/DataAccessLayer/Database/dbContext.cs (hunk 2 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Database/dbContext.cs]: alright

File: Backend/ManagerLayer/KFC_SSO_Utility/KFC_SSO_Manager.cs [hunk 6]
```diff
@@ -41,11 +33,6 @@ public async Task<Session> LoginFromSSO(string Username, Guid ssoID, long timest
             var validSignature = _ssoServiceAuth.IsValidClientRequest(ssoID.ToString(), Username, timestamp, signature);
             if (!validSignature)
             {
```
Comment (alfredovargasjr) [inline Backend/ManagerLayer/KFC_SSO_Utility/KFC_SSO_Manager.cs:35]: is this not being logged anymore?

File: Backend/ManagerLayer/Login/UserLoginManager.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
+﻿using DataAccessLayer.Database;
```
Comment (alfredovargasjr) [inline Backend/ManagerLayer/Login/UserLoginManager.cs]: this no longer exists

File: Backend/ManagerLayer/User/UserManager.cs [hunk 8]
```diff
@@ -15,24 +13,16 @@ namespace ManagerLayer.Users
...
 
         public UserManager(DatabaseContext db)
         {
             _db = db;
-            _loggingManager = new LoggingManager();
         }
 
         public async Task<Session> Login(User user)
         {
             var _authorizationManager = new AuthorizationManager(_db);
```
Comment (alfredovargasjr) [inline Backend/ManagerLayer/User/UserManager.cs:24]: Logging is not done here anymore?

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Backend/DTO/Constants.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            None = 0,
+            Registration,
+            Logout,
+            Login,
+            Mapview,
+            PointDetails,
+            AdminDash,
+            PointEditor,
+            Session,
+            SSO
```
Comment (alfredovargasjr) [inline Backend/DTO/Constants.cs]: There's two of these

File: Backend/DTO/DTO/LoginManagerDTOs.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (alfredovargasjr) [inline Backend/DTO/DTO/LoginManagerDTOs.cs]: Did you make this?

File: Backend/DataAccessLayer/Database/dbContext.cs [hunk 11]
```diff
@@ -12,11 +12,12 @@ namespace DataAccessLayer.Database
...
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Database/dbContext.cs]: Dont forget about this

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

File: Backend/DTO/DTO/LoginManagerDTOs.cs (hunk 10 above)
Comment (ChrisMeyer7088) [reply Backend/DTO/DTO/LoginManagerDTOs.cs]: Nope, should I remove it?

File: Backend/DataAccessLayer/Database/dbContext.cs (hunk 11 above)
Comment (ChrisMeyer7088) [reply Backend/DataAccessLayer/Database/dbContext.cs]: Yup will do

File: Backend/DTO/DTO/LoginManagerDTOs.cs (hunk 10 above)
Comment (alfredovargasjr) [reply Backend/DTO/DTO/LoginManagerDTOs.cs]: Lol idk where this came from. Yea we dont need it

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

**Review by alfredovargasjr — APPROVED** [review]

**Pull request merged**